aiohttp/multipart.py

```diff
@@ -957,6 +955,11 @@ def size(self) -> Optional[int]:
     async def write(self, writer: Any, close_boundary: bool = True) -> None:
         """Write body."""
         for part, encoding, te_encoding in self._parts:
+            if self._is_form_data:
+                # https://datatracker.ietf.org/doc/html/rfc7578#section-4.2
+                assert CONTENT_DISPOSITION in part.payload.headers
```

Comment:
It's still breaks compatibility for code without `rfc7578` proper support.

I would recommend making a `warn` and enable assert from `v4`

Reply:
This resolves your issue. I'll make some additional changes afterwards to resolve #8326.